### PR TITLE
Mention in RTE (but without the pill yet)

### PR DESCRIPTION
**Pull request by Velin92**
Now tapping on a suggestion makes the suggestion appear in RTE as a link, however this is not yet rendered as a Pill, which I will do in the next PR.

https://github.com/vector-im/element-x-ios/assets/34335419/387e2dab-678d-496d-9c90-1220efd5db6e

### CODE REVIEW

File: ElementX/Sources/Screens/ComposerToolbar/ComposerToolbarViewModel.swift [hunk 1]
```diff
@@ -22,6 +22,7 @@ import WysiwygComposer
 typealias ComposerToolbarViewModelType = StateStoreViewModel<ComposerToolbarViewState, ComposerToolbarViewAction>
 
 final class ComposerToolbarViewModel: ComposerToolbarViewModelType, ComposerToolbarViewModelProtocol {
+    private let appSettings: AppSettings
```
Comment (pixlwave) [inline ElementX/Sources/Screens/ComposerToolbar/ComposerToolbarViewModel.swift:25]: Tiny nit: Can we keep these ordered to match the init?

File: ElementX/Sources/Screens/ComposerToolbar/ComposerToolbarViewModel.swift [hunk 2]
```diff
@@ -38,7 +39,8 @@ final class ComposerToolbarViewModel: ComposerToolbarViewModelType, ComposerTool
 
     private var currentLinkData: WysiwygLinkData?
 
-    init(wysiwygViewModel: WysiwygComposerViewModel, completionSuggestionService: CompletionSuggestionServiceProtocol, mediaProvider: MediaProviderProtocol) {
+    init(wysiwygViewModel: WysiwygComposerViewModel, completionSuggestionService: CompletionSuggestionServiceProtocol, mediaProvider: MediaProviderProtocol, appSetting: AppSettings) {
+        appSettings = appSetting
```
Comment (pixlwave) [inline ElementX/Sources/Screens/ComposerToolbar/ComposerToolbarViewModel.swift:43]: Same here

File: ElementX/Sources/Screens/ComposerToolbar/View/CompletionSuggestionView.swift [hunk 3]
```diff
@@ -63,7 +64,9 @@ struct CompletionSuggestionView: View {
 
     private func list() -> some View {
```
Comment (pixlwave) [inline ElementX/Sources/Screens/ComposerToolbar/View/CompletionSuggestionView.swift:65]: Totally unrelated, but shouldn't this be a var?

File: ElementX/Sources/Screens/RoomScreen/RoomScreenCoordinator.swift [hunk 4]
```diff
@@ -59,15 +60,18 @@ final class RoomScreenCoordinator: CoordinatorProtocol {
...
 
         wysiwygViewModel = WysiwygComposerViewModel(minHeight: ComposerConstant.minHeight,
                                                     maxCompressedHeight: ComposerConstant.maxHeight,
                                                     maxExpandedHeight: ComposerConstant.maxHeight,
                                                     parserStyle: .elementX)
-        composerViewModel = ComposerToolbarViewModel(wysiwygViewModel: wysiwygViewModel, completionSuggestionService: parameters.completionSuggestionService, mediaProvider: parameters.mediaProvider)
+        composerViewModel = ComposerToolbarViewModel(wysiwygViewModel: wysiwygViewModel,
+                                                     completionSuggestionService: parameters.completionSuggestionService,
+                                                     mediaProvider: parameters.mediaProvider,
+                                                     appSetting: parameters.appSettings)
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/RoomScreenCoordinator.swift:74]:
```suggestion
                                                     appSettings: parameters.appSettings)
```

**Review by pixlwave — APPROVED** [review]
Oooo completions! 😎

File: ElementX/Sources/Screens/ComposerToolbar/View/CompletionSuggestionView.swift (hunk 3 above)
Comment (Velin92) [reply ElementX/Sources/Screens/ComposerToolbar/View/CompletionSuggestionView.swift:65]: I guess... this code was copied from El-iOS to match the same intrinsic behaviour list + max height based on number of cell from there, so didn't look too much into improving it. Also this is probably going to be completely remade by Design, so is likely going to completely change in the future. This is just to have something that works that can be tested functionality wise

**Pull request merged**